src/main/scala/org/dia/apps/AccumulationsApp.scala

```diff
@@ -0,0 +1,126 @@
+
+object AccumulationsApp extends App {
+  val logger = org.slf4j.LoggerFactory.getLogger(this.getClass)
+  val masterURL = if (args.isEmpty) "local[*]" else args(0)
+  val partitions = if (args.length <= 1) 8 else args(1).toInt
+  val hdfsURL = if (args.length <= 2) "hdfs://localhost:9000" else args(2)
+  val varname = if (args.length <= 3) "precipitation" else args(3)
+  val outputLoc = if (args.length <= 4) "output" else args(4)
+  val credentialsFilePath = "src/test/resources/TestHTTPCredentials"
+  val fileSepStr = System.getProperty("file.separator")
+  val credentialList = Source.fromFile(credentialsFilePath)
+    .getLines()
+    .map(p => {
+      val split = p.split("\\s+")
+      (split(0), split(1))
+    }).toList
+  val username = credentialList(0)._2
+  val password = credentialList(1)._2
+
+  val outputDir = FileUtils.checkHDFSWrite(outputLoc)
+
+  val hdfsDir = outputDir + fileSepStr + "TRMM3B42.txt"
+  val ugi = UserGroupInformation.getCurrentUser().getUserName()
+  val hdfsPath = hdfsURL + fileSepStr + "user" + fileSepStr + ugi + fileSepStr +
+    outputDir + fileSepStr
+
+  logger.info("Starting Simple Accumulation")
+
+  val sc = new SciSparkContext(masterURL, "SciSpark example: Simple Accumulation")
+
+  sc.addHTTPCredential("http://disc2.nascom.nasa.gov/opendap/hyrax/TRMM_L3/", username, password)
+  /**
+   * The data is 3hrly TRMM data. See
+   * http://disc.sci.gsfc.nasa.gov/precipitation/documentation/TRMM_README/TRMM_3B42_readme.shtml
+   * for more details
+   */
+  OpenDapTRMMURLGenerator.run(false, hdfsURL, hdfsDir, "201006150000", "201006151200", 2,
+    List("precipitation,1,1439,1,399", "nlon,1,1439", "nlat,1,399"))
+
+  val sRDD = sc.sciDatasets(hdfsDir, List(varname), partitions)
+
+  /**
+   * Add the frame as an attribute to the SciDataset
+   */
+  val labels = sRDD.map(p => {
+    val splitName = p.datasetName.split("\\.")
+    val FrameID = splitName(1) + splitName(2)
+    p("FRAME") = FrameID
+  })
+  /**
+   * Convert the data from hrly rate every 3 hrs to 3hrly accumulations.
+   */
+  val accu3hrly = labels.map(p => {
+    /* Verbose way to add a new Variable to a sciDataset */
+    val precipAccStr = "(" + varname + " * 3.0)"
+    val precipAccTensor = p(varname)() * 3.0
+    val precipAccVar = new Variable("precip3hrlyAcc", "float", precipAccTensor.data,
+      p(varname).shape(), List(("Calculation", precipAccStr)), p(varname).dims)
+    precipAccVar.insertAttributes(("long_name", "precipitation 3hrlyAccumulation"))
+    precipAccVar.insertAttributes(("units", "mm"))
+    p.insertVariable(("precip3hrlyAcc", precipAccVar))
+  })
+
+  /**
+   * To find 6hrly accumulation, first the time must be maintained.
+   */
+  val consecutiveTimes = SRDDFunctions.fromRDD(accu3hrly).pairConsecutiveFrames("FRAME")
+
+  /**
+   * Sum consecutive frames for 6hrly accumulation
+   */
+  val accu6hrly = consecutiveTimes.map(p => {
+    /* Concise way to add a new Variable to the sciDataset using the update function */
```

Comment:
Since we're using p._1 and p._2 to access the tuple elements, how do you feel about using case statements to unpack the tuple?

Similar to the following map function found here:
https://github.com/SciSpark/SciSpark/blob/master/src/main/scala/org/dia/algorithms/mcs/MCSOps.scala#L360